### ISSUE TITLE
reproducing your result

### ISSUE THREAD

**Issue by NoaGarnett** [WeiChengTseng/PCGrad#13]
Thanks for the simple and elegant implementation!
I tried running your code as is, on Multi-MNIST data, and failed to reproduce results.
I ran main_multi_mnist.py without changing any  hyper parameter (learning rate (0.0005) , batch size (256), number of epochs (100)). For comparison, I created a version with no pcgrad:
     1. comment out line 57: ## optimizer = PCGrad(optimizer)
     2. replace line 72: optimizer.pc_backward(losses) -> torch.sum(torch.stack(losses)).backward()

I run each version 7 times. my results (averaging left-digit and right-digit accuracy) are:

Without PCGrad: average accuracy 89.5%, max accuracy 89.9%, standard deviation 0.38
With      PCGrad: average accuracy 89.5%, max accuracy 89.8%, standard deviation 0.20

Can you come up with an explanation?
Many thanks,
Noa Garnett

**Comment by Lizhaoqing123** [comment]
> Thanks for the simple and elegant implementation! I tried running your code as is, on Multi-MNIST data, and failed to reproduce results. I ran main_multi_mnist.py without changing any hyper parameter (learning rate (0.0005) , batch size (256), number of epochs (100)). For comparison, I created a version with no pcgrad: 1. comment out line 57: ## optimizer = PCGrad(optimizer) 2. replace line 72: optimizer.pc_backward(losses) -> torch.sum(torch.stack(losses)).backward()
> 
> I run each version 7 times. my results (averaging left-digit and right-digit accuracy) are:
> 
> Without PCGrad: average accuracy 89.5%, max accuracy 89.9%, standard deviation 0.38 With PCGrad: average accuracy 89.5%, max accuracy 89.8%, standard deviation 0.20
> 
> Can you come up with an explanation? Many thanks, Noa Garnett

My result is same as yours, did you finally solve this problem?

**Comment by NoaGarnettDia** [comment]
No, never did. Just moved on...

**Comment by yangmin666** [comment]
> 

Have you solved it? I have the same issue.

**Comment by NoaGarnett** [comment]
No, gave it up long ago.

**Comment by lvilaca16** [comment]
Hi,
I refactored the entire repository and added a new PCGrad version to my [fork](https://github.com/lvilaca16/Pytorch-PCGrad). I hope this helps!
Unfortunately, the results are not as expected. 

Without PCGrad: val_loss = 1.04, left_acc = 91.26, right_acc = 89.03
PCGrad (mean reduction): val_loss = 1.02, left_acc = 91.58, right_acc = 89.35

All these results are averages of 10 runs.
